server/src/test/java/org/elasticsearch/cluster/routing/allocation/FailedShardsRoutingTests.java

```diff
@@ -717,13 +717,13 @@ public void testReplicaOnNewestVersionIsPromoted() {
                     .add(
                         newNode(
                             "node3-old",
-                            VersionUtils.randomVersionBetween(random(), Version.CURRENT.minimumIndexCompatibilityVersion(), null)
+                            VersionUtils.randomVersionBetween(random(), Version.CURRENT.minimumCompatibilityVersion(), null)
```

Comment:
This doesn't seem equivalent. This test seems to want to randomize back to the oldest supported node version, as far as index compatibility is concerned, not transport (which is what minimumCompatibilityVerison is right? or at least that's what it was before creating TransportVersion)

Reply:
But the nodes can only form a cluster if its at least `minimumCompatibleVersion`, no?

Reply:
I've modified it to specify the node & index versions separately